crates/indexer/src/models/token_bcs_utils.rs

```diff
@@ -0,0 +1,31 @@
+use serde::{Serialize, Deserialize, Deserializer};
+use serde::de::Error;
+
+pub fn deserialize_bcs_from_string<'de, D>(deserializer: D) -> Result<String, D::Error>
+    where
+        D: Deserializer<'de>,
+{
+    // convert string to bytes
+    let s = <String>::deserialize(deserializer)?;
+    let decoded = hex::decode(s).expect("Hex string to bcs decoding failed");
```

Comment:
Let's propagate error:
```
let decoded = hex::decode(s).map_err(D::Error::custom)?;
```

Reply:
If deserialization fails, will return the original string back to not stop the deserialization. You can also add your customized log in this function